src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5829,6 +5831,39 @@ public void onEviction(Sha1HashKey key, PreparedStatementHandle handle) {
             }
         }
     }
+
+    boolean isAzureDW() throws SQLServerException, SQLException {
+        if (null == isAzureDW) {
+            try (Statement stmt = this.createStatement(); ResultSet rs = stmt.executeQuery("SELECT CAST(SERVERPROPERTY('EngineEdition') as INT)");)
```

Comment:
We don't seem to catch/handle any exceptions from this try statement, is there a reason why we're using it here.

Reply:
Normally, when we create a new Statement or ResultSet object, it's good practice to close it. However, the try-with-resources syntax (which i've used here) allows those objects to get deleted automatically. You can read more on this page: https://blogs.oracle.com/weblogicserver/using-try-with-resources-with-jdbc-objects